strict type hint (#114)
Pycord-Development/pycord-v3: .github/CONTRIBUTING.md

# Pycord-v3 Contribution Guide

This is an extremely basic guide for contributing to the repo.

## Steps for Every Contribution

***There is only 1.***

1. Make sure your contribution is not a duplicate before opening (i.e., search all open contributions and ensure it doesn't closely relate to yours).

## Steps for Opening Issues

Thanks for taking the time to report an issue, this helps us make a more stable and usable library.

> 1. If your issue produces a traceback/exception, include the entirety of it. Please put it in a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks).

Reviewer: ```suggestion
1. If your issue produces an exception, include the entire traceback in a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks).
```
Exceptions always leave tracebacks, unless it's handled by another piece of code.

> 2. Fill out the template. If the template is not filled out properly, your issue will most likely be closed.

Reviewer: Isn't that what step 1 is telling you to do, but with the traceback only?

Reviewer 2: Somewhat, but the directions on how to make a code block should probably be put somewhere.

3. Be as detailed as possible when creating your bug report. In most scenarios, the fewer details that we are given, the longer it will take for your issue to be resolved.

## Steps for Opening Pull Requests

Considering that we are in a pre-alpha state right now, thank you very much for your contribution. It is deeply appreciated.

However, before you create your PR, please ensure you've done a couple things:

1. Your PR needs to have a clear purpose. PRs that edit multiple files are more than welcome, but make sure it accomplishes a single goal. This could be fixing a bug, refactoring, or even adding new material.
2. Follow the style of the library. This implies two separate points:

>    1. Make sure your code is as clean as possible and looks similar to the rest of the codebase. This includes spacing, function signatures/definitions, and commenting/documentation format.

Reviewer: Encourage usage of pre-commit as well, since those hooks tackle formatting.

Reviewer 2: Not sure how that would look tbh with you, mind writing an example?
